Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/ApplePayTokenizationViewModel.swift

```diff
@@ -314,6 +319,53 @@ class ApplePayTokenizationViewModel: PaymentMethodTokenizationViewModel {
     }
 }
 
+@available(iOS 11.0, *)
+extension ApplePayTokenizationViewModel {
+    
+    private func clientSessionBillingAddressFromApplePayBillingContact(_ billingContact: PKContact?) -> ClientSession.Address? {
+        
+        guard let postalAddress = billingContact?.postalAddress else {
+            return nil
+        }
+        
+        // From: https://developer.apple.com/documentation/contacts/cnpostaladdress/1403414-street
+        let addressLines = postalAddress.street.components(separatedBy: "\n")
+        let addressLine1 = addressLines.first
+        let addressLine2 = addressLines.count > 1 ? addressLines[1] : nil
+        
+        return ClientSession.Address(firstName: billingContact?.name?.givenName,
+                                     lastName: billingContact?.name?.familyName,
+                                     addressLine1: addressLine1,
+                                     addressLine2: addressLine2,
+                                     city: postalAddress.city,
+                                     postalCode: postalAddress.postalCode,
+                                     state: postalAddress.state,
+                                     countryCode: CountryCode(rawValue: postalAddress.isoCountryCode))
+    }
+    
+    private func updateBillingAddressViaClientSessionActionWithAddress(_ address: ClientSession.Address?) -> Promise<Void> {
```

Comment:
Should it even get in here if `address` is `nil`?

Reply:
well now that we have the `...IfNeeded` we clarify whether te Promise should be performed or not